lib/Promise.php

```diff
@@ -251,7 +256,11 @@ private function invokeCallback(Promise $subPromise, callable $callBack = null):
                 if (self::FULFILLED === $this->state) {
                     $subPromise->fulfill($this->value);
                 } else {
-                    $subPromise->reject($this->value);
+                    if ($this->value instanceof \Throwable) {
+                        $subPromise->reject($this->value);
+                    } else {
+                        throw new \LogicException('The subPromise was not fulfilled but no exception was specified');
+                    }
```

Comment:
phpstan is confident that "value" will be a Throwable at this point. So I added an extra check, and throw a LogicException if value was not a Throwable. LogicException is already used elsewhere in this code, so I chose to throw that.